bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -634,11 +645,29 @@ pub mod pallet {
 				config::SLOTS_PER_EPOCH as u64,
 			));
 			let domain_type = config::DOMAIN_SYNC_COMMITTEE.to_vec();
-			// Domains are used for for seeds, for signatures, and for selecting aggregators.
+			// Domains are used for seeds, for signatures, and for selecting aggregators.
 			let domain = Self::compute_domain(domain_type, fork_version, validators_root)?;
 			// Hash tree root of SigningData - object root + domain
 			let signing_root = Self::compute_signing_root(header, domain)?;
 			Ok(signing_root)
 		}
+
+		/// Updates are free if the update is successful and the interval between the latest
+		/// finalized header in storage and the newly imported header is large enough. All
+		/// successful sync committee updates are free.
+		pub(super) fn check_refundable(update: &Update, latest_slot: u64) -> Pays {
+			// If the sync committee was successfully updated, the update may be free.
+			if update.next_sync_committee_update.is_some() {
+				return Pays::No;
+			}
+
+			// If free headers are allowed and the latest finalized header is larger than the
+			// minimum slot interval, the header import transaction is free.
+			if update.finalized_header.slot >= latest_slot + T::FreeHeadersInterval::get() as u64 {
```

Comment:
Should use saturating arithmetic to prevent panic.

Reply:
Added in https://github.com/paritytech/polkadot-sdk/pull/5201/commits/9391a36368e962f05a1987bff7e9dda82a54b9cf.